sample/src/main/java/org/walletconnect/example/wallet/scanner/ScannerFragment.kt

```diff
@@ -0,0 +1,143 @@
+package org.walletconnect.example.wallet.scanner
+
+import android.Manifest
+import android.annotation.SuppressLint
+import android.content.pm.PackageManager
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.camera.core.CameraSelector
+import androidx.camera.core.ImageAnalysis
+import androidx.camera.core.ImageProxy
+import androidx.camera.core.Preview
+import androidx.camera.lifecycle.ProcessCameraProvider
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.activityViewModels
+import com.google.mlkit.vision.barcode.Barcode
+import com.google.mlkit.vision.barcode.BarcodeScanner
+import com.google.mlkit.vision.barcode.BarcodeScannerOptions
+import com.google.mlkit.vision.barcode.BarcodeScanning
+import com.google.mlkit.vision.common.InputImage
+import org.walletconnect.example.R
+import org.walletconnect.example.databinding.ScannerFragmentBinding
+import org.walletconnect.example.wallet.WalletViewModel
+import java.util.concurrent.Executors
+
+class ScannerFragment : Fragment(R.layout.scanner_fragment) {
+
+    private lateinit var binding: ScannerFragmentBinding
+    private val viewModel: WalletViewModel by activityViewModels()
+
+    private var cameraProvider: ProcessCameraProvider? = null
+    private var cameraSelector: CameraSelector? = null
+    private var previewUseCase: Preview? = null
+    private var analysisUseCase: ImageAnalysis? = null
+    private var shouldScan: Boolean = true
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding = ScannerFragmentBinding.bind(view)
+        activity?.invalidateOptionsMenu()
+        setupCamera()
+        viewModel.hideBottomNav()
+    }
+
+    private fun setupCamera() {
+        cameraSelector =
+            CameraSelector.Builder().requireLensFacing(CameraSelector.LENS_FACING_BACK).build()
+        setupObserver()
+    }
+
+    private val cameraPermissionCallback =
+        registerForActivityResult(ActivityResultContracts.RequestPermission()) { isGranted ->
+            if (isGranted) {
+                bindCameraUseCases()
+            }
+        }
+
+    private fun setupObserver() {
+        viewModel
+            .processCameraProvider
+            .observe(viewLifecycleOwner, { provider: ProcessCameraProvider ->
+                cameraProvider = provider
+                if (isCameraPermissionGranted()) {
+                    bindCameraUseCases()
+                } else {
+                    cameraPermissionCallback.launch(Manifest.permission.CAMERA)
+                }
+            })
+    }
```

Comment:
If the CameraProvider LD doesn't have to be in the ViewModel, then this block of code will be replaced by 
```suggestion
    private fun setupObserver() {
        getProcessCameraProvider()
            .observe(viewLifecycleOwner, { provider: ProcessCameraProvider ->
                cameraProvider = provider
                if (isCameraPermissionGranted()) {
                    bindCameraUseCases()
                } else {
                    cameraPermissionCallback.launch(Manifest.permission.CAMERA)
                }
            })
    }

    private fun getProcessCameraProvider(): LiveData<ProcessCameraProvider> {
        val cameraProviderLiveData = MutableLiveData<ProcessCameraProvider>()
        val cameraProviderFuture = ProcessCameraProvider.getInstance(getApplication())
        cameraProviderFuture.addListener(
            { cameraProviderLiveData.setValue(cameraProviderFuture.get()) },
            ContextCompat.getMainExecutor(requireActivity().application)
        )
        return cameraProviderLiveData
    }
```